### PR TITLE
mgr/volumes: Bump up AuthMetadataManager's version

### PR DESCRIPTION
**Pull request by kotreshhr**
With ceph_volume_client and mgr-volumes co-existing
for sometime, the version of both needs to be same.
The ceph_volume_client version <=5 can't decode
'subvolumes' key in auth-metadata file. Hence to
handle version in-compatibility, the version of
ceph_volume_client is bumped up to 6 and the same
needs to be done in mgr-volume's AuthMetadataManager.

Fixes: https://tracker.ceph.com/issues/49374
Signed-off-by: Kotresh HR <khiremat@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py [hunk 1]
```diff
@@ -203,6 +203,6 @@ def subvol_metadata_set(self, group_name, subvol_name, data):
         decode the metadata and 'version', the AuthMetadataManager version
         that encoded the metadata.
         """
-        data['compat_version'] = 1
+        data['compat_version'] = 6
```
Comment (ajarr) [inline src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py:206]: instead do this in  auth_metadata_set()?

**Review by ajarr — APPROVED** [review]

File: src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py (hunk 1 above)
Comment (ajarr) [reply src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py:206]: instead do this in auth_metadata_set()?

**Review by ajarr — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py (hunk 1 above)
Comment (kotreshhr) [reply src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py:206]: Oh yes, right!

Comment (kotreshhr) [reply src/pybind/mgr/volumes/fs/operations/versions/auth_metadata.py]: Done

**Review by ajarr — APPROVED** [review]

**Comment by kotreshhr** [comment]
Teuthology Results:

http://pulpito.front.sepia.ceph.com/khiremat-2021-02-22_06:49:11-fs:volumes-wip-khiremat-39572-3-distro-basic-smithi/

**Pull request merged**